src/main/java/net/dv8tion/jda/api/entities/Guild.java

```diff
@@ -316,161 +316,26 @@ default RestAction<Void> deleteCommandById(long commandId)
-     * To change this behavior, use {@link CommandData#setDefaultEnabled(boolean)} on your command.
-     *
-     * <p>These privileges are used to restrict who can use commands through Role/User whitelists/blacklists.
-     *
-     * <p>If there is no command with the provided ID,
-     * this RestAction fails with {@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_COMMAND ErrorResponse.UNKNOWN_COMMAND}
-     *
-     * @param  id
-     *         The id of the command, this can be global or guild command
-     * @param  privileges
-     *         Complete list of up to 10 {@link CommandPrivilege CommandPrivileges} for this command
-     *
-     * @throws IllegalArgumentException
-     *         If null is provided or more than 10 privileges are provided
-     *
-     * @return {@link RestAction} - Type: {@link List} or {@link CommandPrivilege}
-     *         The updated list of privileges for this command.
-     */
-    @Nonnull
-    @CheckReturnValue
-    default RestAction<List<CommandPrivilege>> updateCommandPrivilegesById(long id, @Nonnull CommandPrivilege... privileges)
-    {
-        Checks.noneNull(privileges, "CommandPrivileges");
-        return updateCommandPrivilegesById(id, Arrays.asList(privileges));
-    }
-
-    /**
-     * Updates the list of {@link CommandPrivilege CommandPrivileges} for the specified commands.
-     * <br>The argument for this function is a {@link Map} similar to the one returned by {@link #retrieveCommandPrivileges()}.
-     * <br>Note that commands are enabled by default for all members of a guild, which means you can only <em>blacklist</em> roles and members using this method.
-     * To change this behavior, use {@link CommandData#setDefaultEnabled(boolean)} on your command.
-     *
-     * <p>These privileges are used to restrict who can use commands through Role/User whitelists/blacklists.
-     *
-     * <p>If there is no command with the provided ID,
-     * this RestAction fails with {@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_COMMAND ErrorResponse.UNKNOWN_COMMAND}
-     *
-     * @param  privileges
-     *         Complete map of {@link CommandPrivilege CommandPrivileges} for each command
-     *
-     * @throws IllegalArgumentException
-     *         If null is provided, any of the map keys is not a valid snowflake, or more than 10 privileges are provided for any command
-     *
-     * @return {@link RestAction} - Type: {@link Map} from {@link String} Command ID to {@link List} of {@link CommandPrivilege}
-     *         The updated map of command privileges for this guild.
-     */
-    @Nonnull
-    @CheckReturnValue
-    RestAction<Map<String, List<CommandPrivilege>>> updateCommandPrivileges(@Nonnull Map<String, ? extends Collection<CommandPrivilege>> privileges);
+    RestAction<Map<String, List<IntegrationPrivilege>>> retrieveCommandPrivileges();
```

Comment:
I think we could offer a better type for this than a map, we could easily take care of application permissions by using the id `jda.getSelfUser().getApplicationId()`.

```java
interface SomethingPrivilegeType {
  List<Privilege> getAppPrivs();
  List<Privilege> getCommandPrivs(id);
  Map<String, List<Privilege>> getAsMap();
}
```

Reply:
Was this ignored?

Reply:
No i just didn't look at it yet